lifecycle/cli.go

```diff
@@ -75,10 +77,19 @@ func validateCreateReleaseBundleContext(c *cli.Context) error {
 		return err
 	}
 
-	bothProvided := c.IsSet(cliutils.Builds) && c.IsSet(cliutils.ReleaseBundles)
-	noneProvided := !c.IsSet(cliutils.Builds) && !c.IsSet(cliutils.ReleaseBundles)
-	if bothProvided || noneProvided {
-		return errorutils.CheckErrorf("exactly one of the following options must be supplied: --%s or --%s", cliutils.Builds, cliutils.ReleaseBundles)
+	return assertValidCreationMethod(c)
+}
+
+func assertValidCreationMethod(c *cli.Context) error {
+	methods := []bool{
+		c.IsSet("spec"), c.IsSet(cliutils.Builds), c.IsSet(cliutils.ReleaseBundles)}
+	if coreutils.SumTrueValues(methods) > 1 {
+		return errorutils.CheckErrorf("exactly one creation source must be supplied: --%s, --%s or --%s.\n"+
+			"The spec option is the recommended approach.", "spec", cliutils.Builds, cliutils.ReleaseBundles)
```

Comment:
If `--builds` and `--release-bundles` are deprecated, I suggest rephrasing "The spec option is the recommended approach" to something like "The --builds and --release-bundles options are deprecated."